coil-compose-base/src/main/java/coil/compose/AsyncImagePainter.kt

```diff
@@ -247,17 +251,19 @@ class AsyncImagePainter internal constructor(
     internal var contentScale = ContentScale.Fit
     internal var filterQuality = DefaultFilterQuality
     internal var isPreview = false
+    internal val requestInternal = MutableStateFlow(request)
 
     /** The current [AsyncImagePainter.State]. */
     var state: State by mutableStateOf(State.Empty)
         private set
 
     /** The current [ImageRequest]. */
-    var request: ImageRequest by mutableStateOf(request)
-        internal set
+    val request
+        @Composable
+        get() = requestInternal.collectAsState().value
 
     /** The current [ImageLoader]. */
-    var imageLoader: ImageLoader by mutableStateOf(imageLoader)
+    var imageLoader: ImageLoader = imageLoader
```

Comment:
I think we need to keep this as a `mutableStateOf` otherwise it violates `@Stable`'s contract. It's possible to change the `ImageLoader` from `rememberAsyncImagePainter` and the composition should be notified.